setup.py

```diff
@@ -1,20 +1,29 @@
 from setuptools import setup, find_packages
+from pip.req import parse_requirements
+import uuid
+
+setup_requirements = [
+    'pytest-runner'
+]
+
+test_requirements = [
+    'pytest'
+]
+
+install_requirements = [str(ir.req) for ir in parse_requirements(
+                        "requirements.txt", session=uuid.uuid1())]
```

Comment:
Is there a good reason to keep `requirements.txt` which makes us have to do clever things to read it into `setup.py`? `pip install -e.` seems to work very well so even for local dev installs we don't need `requirements.txt` anymore. I'd vote for removing it.

Reply:
Yea, this is a good question. I guess I added one because most python modules I've seen have one, and because having one enables (free!) dependency version analysis tools like https://pyup.io to work with the project. (On a related note, I would vote for pinning the requirements to particular version numbers at some point, but thought that might merit a separate PR).

Reply:
This is an anti-pattern that will break in the next pip release.  pip doesn’t have a stable Python API, so using this kind of code is risky (and will be broken on purpose in pip 10).  It’s also confusing and a bit backward: setup.py was defined first (and can contain open requirements, i.e. without too strict version constraints, to maximize compatibility with other libraries), requirement files came up after (and these are useful with pinned versions to define known good sets for repeatable installs).

Links:
https://caremad.io/posts/2013/07/setup-vs-requirement/
http://nvie.com/posts/better-package-management/

A pretty great solution these days is to have unversioned dependencies in setup.py and use pip-tools to generate a requirements file with exact versions.

(Test dependencies don’t have to be in setup.py, they can be in their own requirements file.)

Reply:
@merwok , I appreciate the background and links! I'll dive deeper into this later :)

Reply:
This is a busy week but maybe the week after I could send you a PR!  Or feel free to tag me in the meantime.